invenio_files_rest/__init__.py

```diff
@@ -6,7 +6,744 @@
+
+Complete a multipart upload, by using the POST command; it will return a 200
+code on success:
+
+.. code-block:: console
+
+   $ curl -i -X POST \
+     "http://localhost:5000/files/$B/my_file.txt?uploadId=$U"
+
+Abort a multipart upload (deletes all uploaded parts); it will return a 204
+code if it succeeds:
+
+.. code-block:: console
+
+   $ curl -i -X DELETE "http://localhost:5000/files/$B/my_file.txt?uploadId=$U"
+
+Large Files
+^^^^^^^^^^^
+
+For smaller files, uploads are stored in the webserver's memory.
+As for larger files, they are stored in a temporary location.
+You will need some extra configurations for nginx and Flask application.
+
+1. There is a possibility that Ngxinx might return
+:code:`413 (Request Entity Too Large)` for large files. In the configuration
+the body size of the request can be customized according to our needs. The
+following example configures nginx to accept up to :code:`25MB`.
+
+.. code-block:: console
+
+    http {
+        ...
+        client_max_body_size 25M;
+    }
+
+2. Specify :code:`MAX_CONTENT_LENGTH` header otherwise Flask will reject
+incoming requests with a content length greater than this by returning a
+:code:`413 (Request Entity Too Large)`. If not set and the request does not
+specify a :code:`CONTENT_LENGTH`, no data will be read for security. You can
+set the :code:`MAX_CONTENT_LENGTH` to :code:`25MB`  like in the example below.
+
+.. code-block:: console
+
+    $ app.config['MAX_CONTENT_LENGTH'] = 25 * 1024 * 1024
+
+.. note::
+
+    Special note on the :code:`get_data()` method: Calling this loads the full
+    request data into memory. This is only safe to do if the
+    :code:`max_content_length` is set.
+
+
+Retrieve Files
+--------------
+Once the bucket is created and a file is uploaded, it is possible
+to retrieve it with a GET request.
+
+By default, the latest version will be downloaded. To retrieve a specific
+version of the file, the versionId query parameter can be used, as in the
+example below:
+
+Download the latest version of the file:
+
+.. code-block:: console
+
+   $ curl -i http://localhost:5000/files/$B/my_file.txt
+
+Download a specific version of the file:
+
+.. code-block:: console
+
+   $ curl -i http://localhost:5000/files/$B/my_file.txt?versionId=<version_id>
+
+**NOTE:** By default, the file is returned with the header
+:code:`'Content-Disposition': 'inline'`, so that the browser will try to
+preview it. In case you want to trigger a download of the file, use the
+:code:`download` query parameter, which will change the
+:code:`'Content-Disposition'` header to :code:`'attachment'`
+
+.. code-block:: console
+
+   $ curl -i http://localhost:5000/files/$B/my_file.txt?download
+
+
+Security
+^^^^^^^^
+It is very easy to be exposed to Cross-Site Scripting (XSS) attacks if you
+serve user uploaded files. Here are some recommendations:
+
+1. Serve user uploaded files from a separate domain (not a subdomain). This
```

Comment:
All the points are rendered without any line brake- maybe we should add newlines between them